api/plugins/action/deploy_target_config.py

```diff
@@ -90,35 +90,42 @@ def run(self, tmp=None, task_vars=None):
         return result
 
 def determine_required_updates(existing_configs, desired_configs):
-    addition_required = []  
-    deletion_required = [
-        config['id']
-        for config in existing_configs
-        if not any(
-            config['branches'] == desired['branches']
-            for desired in desired_configs
-        )
-    ]
-
+    addition_required = []
+    deletion_required = []
+
+    grouped_configs = {}
+    for config in existing_configs:
+        key = (config['branches'], config['pullrequests'], str(config['deployTarget']['id']), str(config['weight']))
```

Comment:
Is `config['weight']` a good field to create a unique key on if this is to fix duplicate configuration items? `weight` is generally used to control ordering and it is common practice to use 0 when no weight is given.

For example:

```
configs = [{
	"id": 3524,
	"branches": "^master$",
	"pullrequests": "false",
	"deployTarget": {
		"created": "2021-10-06 05:05:44",
		"id": 6
	},
	"weight": 1
},
{
	"id": 3525,
	"branches": "^master$",
	"pullrequests": "false",
	"deployTarget": {
		"created": "2021-10-06 05:05:44",
		"id": 6
	},
	"weight": 2
}
]
```

The key would look something like:

```
Created key: ('^master$', 'false', '6', '1')
Created key: ('^master$', 'false', '6', '2')
```

For each item and it would skip removing the duplicate entry for `^master$`